tests/test_version.py

```diff
@@ -0,0 +1,95 @@
+"""Test Version object and cast_version utility function."""
+import pytest
+from pydantic import ValidationError
+
+from polus.plugins._plugins.io import Version
+from polus.plugins._plugins.utils import cast_version
+
+G = [
+    "1.2.3",
+    "1.4.7-rc1",
+    "4.1.5",
+    "12.8.3",
+    "10.2.0",
+    "2.2.3-dev5",
+    "0.3.4",
+    "0.2.34-rc23",
+]
+B = ["02.2.3", "002.2.3", "1.2", "1.0", "1.03.2", "23.3.03", "d.2.4"]
+
+
+@pytest.mark.parametrize("ver", G, ids=G)
+def test_version(ver):
+    """Test Version pydantic model."""
+    Version(version=ver)
+
+
+@pytest.mark.parametrize("ver", G, ids=G)
+def test_cast_version(ver):
+    """Test cast_version utility function."""
+    cast_version(ver)
+
+
+@pytest.mark.parametrize("ver", B, ids=B)
+def test_bad_version1(ver):
+    """Test ValidationError is raised for invalid versions."""
+    with pytest.raises(ValidationError):
+        cast_version(ver)
+
+
+mj = ["2.4.3", "2.98.28", "2.1.2", "2.0.0", "2.4.0"]
+mn = ["1.3.3", "7.3.4", "98.3.12", "23.3.0", "1.3.5"]
```

Comment:
same comment as before. I start to get used to it but still, identical_major_versions, identical_minor_versions or whatever name reflect your intentions better go a long way to make your code more readable.

Reply:
Solved